cmd/config/config.go

```diff
@@ -0,0 +1,285 @@
+package config
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/manifoldco/promptui"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v3"
+
+	"github.com/zitadel/zitadel/internal/api/authz"
+	"github.com/zitadel/zitadel/internal/api/grpc/oidc/v2"
+	"github.com/zitadel/zitadel/internal/api/saml"
+	"github.com/zitadel/zitadel/internal/command"
+	"github.com/zitadel/zitadel/internal/config/network"
+	"github.com/zitadel/zitadel/internal/database"
+)
+
+// ANSI color codes
+const (
+	colorReset  = "\033[0m"
+	colorBold   = "\033[1m"
+	colorGreen  = "\033[32m"
+	colorYellow = "\033[33m"
+	colorCyan   = "\033[36m"
+)
+
+type Config struct {
+	ExternalPort    uint16
+	ExternalDomain  string
+	ExternalSecure  bool
+	Database        database.Config
+	TLS             network.TLS
+	SystemAPIUsers  map[string]*authz.SystemAPIUser
+	OIDC            oidc.Config
+	SAML            saml.Config
+	DefaultInstance command.InstanceSetup
+}
+
+func New() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "config",
+		Short: "manage configuration",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			/*			stdin, err := io.ReadAll(os.Stdin)
+						if err != nil {
+							return fmt.Errorf("Error reading from stdin: %v", err)
+						}*/
+			node := yaml.Node{
+				Kind: yaml.DocumentNode,
+				Content: []*yaml.Node{
+					{
+						Kind: yaml.MappingNode,
+					},
+				},
+			}
+			/*			if err := yaml.Unmarshal(stdin, &node); err != nil {
+						return fmt.Errorf("Error unmarshaling YAML: %v", err)
+					}*/
+			var cfg Config
+			if err := node.Content[0].Decode(&cfg); err != nil {
+				return fmt.Errorf("Error decoding YAML to Config: %v", err)
+			}
+			note, err := addSystemUser(&cfg, node.Content[0])
+			if err != nil {
+				return fmt.Errorf("Error adding System user: %v", err)
+			}
+			result, err := yaml.Marshal(&node)
+			if err != nil {
+				return fmt.Errorf("Error encoding Config to YAML: %v", err)
+			}
+			_, err = os.Stdout.Write(result)
+			if err != nil {
+				return fmt.Errorf("Error writing YAML to stdout: %v", err)
+			}
+
+			// Print follow-up instructions to stderr with colors
+			os.Stderr.Write([]byte("\n" + colorCyan + strings.Repeat("=", 80) + colorReset + "\n"))
+			os.Stderr.Write([]byte(colorBold + colorGreen + "NEXT STEPS" + colorReset + "\n"))
+			os.Stderr.Write([]byte(colorCyan + strings.Repeat("=", 80) + colorReset + "\n\n"))
+
+			os.Stderr.Write([]byte(colorBold + "1. Generate a master key and store it securely:" + colorReset + "\n\n"))
+			os.Stderr.Write([]byte("   " + colorYellow + "$ zitadel config generate-masterkey > masterkey.txt" + colorReset + "\n\n"))
```

Comment:
I ran the setup with `go main.go config > ./config.yaml`. This message is then misleading because there is no `zitadel` binary generated.

  - If we compile the binary first, it should display `zitadel config generate-masterkey > masterkey.txt`
  - If we run zitadel with `go run` then it should display `go run main.go config generate-masterkey > masterkey.txt`